### DISCUSSION TITLE
Question: How to use log when set window.log = log.functions

### DISCUSSION THREAD

**Discussion by MonsieurPatate** [megahertz/electron-log#290] [Q&A] [answered]
So from [here](https://github.com/megahertz/electron-log/blob/HEAD/docs/node-integration.md) I chose the first method and didin't queit understand how to call log?)

I've tried in renderer process:
```
window.log.error('some error');

console.error('some error'
```
How to call the `electron-log` using the first aproach?
Any help please?)

Thanks

**Comment by megahertz** [comment, marked as answer]
Depending on Electron version and webPreferences, if `window.log = log.functions` doesn't works, you may use `contextBridge.exposeInMainWorld('log', log.functions)`;

**Reply by MonsieurPatate** [reply]
it's a kind of magic =)
Thanks

**Reply by petestarkey** [reply]
Thanks, worked for me in electron-react-boilerplate. It would be helpful if this was mentioned in the docs.

Also, in case this helps anyone, I realised that the renderer outputs to a separate log file to the main process, so you will end up with two log files.

**Reply by Brilliant2412000** [reply]
> Thanks, worked for me in electron-react-boilerplate. It would be helpful if this was mentioned in the docs.
> 
> Also, in case this helps anyone, I realised that the rederer outputs to a separate log file to the main process, so you will end up with two log files.

how to you use in renderer process with electron-react-boilerplate I use but can't see log error file and how to use in preload.js and App.tsx to generate file error

**Reply by petestarkey** [reply]
Put this in preload.js:

`const log = require('electron-log');`
`contextBridge.exposeInMainWorld('log', log.functions);`

And put this in App.tsx:

`window.log.error('Error, from renderer side!');` 

Or this in App.tsx if you want to log standard console errors:

`console.error = window.log.error;` 
`console.error("test console error")`

And then once you run your app, you should get a file called _renderer.log_ in _C:\Users\[username]\AppData\Roaming\BadgePrintingKiosk\logs_ showing your test error.

**Reply by Brilliant2412000** [reply]
> 

i use this template https://electron-react-boilerplate.js.org/
I add this code into AppUpdater Constructor in main.ts and have main.log in folder with path project but I can't have renderer.log in this folder path. It log in terminal when I run it. How can I fix it
```
log.transports.file.resolvePath = (variables) => {
      return process.env.NODE_ENV === 'production'
        ? path.join(
            process.env.PORTABLE_EXECUTABLE_DIR,
            `${variables.appName}/${variables.fileName}`
          )
        : path.join(
            __dirname,
            `../../${variables.appName}/${variables.fileName}`
          );
    };
```
